src/librustc/ty/query/plumbing.rs

```diff
@@ -465,21 +465,17 @@ impl<'tcx> TyCtxt<'tcx> {
 
         let result = if let Some(result) = result {
             profq_msg!(self, ProfileQueriesMsg::CacheHit);
-            self.sess.profiler(|p| p.record_query_hit(Q::NAME));
```

Comment:
We don't need this event?

Reply:
I was pondering this question too. I opted to remove it because we already record the `incr_cache_loading` event above and strictly speaking this is not an in-memory cache hit. However, I also want to replace the `query_provider` event right below with something that indicates that this is not a regular query provider invocation but was triggered instead of a cache load.

Reply:
Ok, sounds good